app/presenters/register_api/single_application_presenter.rb

```diff
@@ -0,0 +1,245 @@
+        array << qualification_to_hash(gcse)
+                     .merge(subject: subject.humanize, grade: hash['grade'], id: hash['public_id'])
+      end
+    end
+
+    def qualification_to_hash(qualification)
+      {
+        id: qualification.public_id,
+        qualification_type: qualification.qualification_type,
+        non_uk_qualification_type: qualification.non_uk_qualification_type,
+        subject: qualification.subject,
+        grade: grade_details(qualification),
+        start_year: qualification.start_year,
+        award_year: qualification.award_year,
+        institution_details: institution_details(qualification),
+        equivalency_details: qualification.composite_equivalency_details,
+        comparable_uk_degree: qualification.comparable_uk_degree,
+      }.merge HesaQualificationFieldsPresenter.new(qualification).to_hash
+    end
+
+    def grade_details(qualification)
+      grade = nil
+
+      if qualification.grade
+        if qualification.predicted_grade
+          grade = "#{qualification.grade} (Predicted)"
+        else
+          grade = qualification.grade
+        end
+      end
+
+      constituent_grades = qualification.constituent_grades
+
+      # For triple award science we need to serialize 'grades' to the 'grade' field
+      # in the specified order
+      if qualification.subject == 'science triple award' && constituent_grades
+        grade = "#{constituent_grades['biology']['grade']}#{constituent_grades['chemistry']['grade']}#{constituent_grades['physics']['grade']}"
+      end
+
+      grade
+    end
+
+    def institution_details(qualification)
+      if qualification.institution_name
+        [qualification.institution_name, qualification.institution_country].compact.join(', ')
+      end
+    end
+
+    def contact_details
+      if application_form.international?
+        address_line1 = application_form.address_line1 || application_form.international_address
+
+        {
+          phone_number: application_form.phone_number,
+          address_line1: address_line1,
+          address_line2: application_form.address_line2,
+          address_line3: application_form.address_line3,
+          address_line4: application_form.address_line4,
+          country: application_form.country,
+          email: application_form.candidate.email_address,
+        }
+      else
+        {
+          phone_number: application_form.phone_number,
+          address_line1: application_form.address_line1,
+          address_line2: application_form.address_line2,
+          address_line3: application_form.address_line3,
+          address_line4: application_form.address_line4,
+          postcode: application_form.postcode,
+          country: application_form.country,
+          email: application_form.candidate.email_address,
+        }
+      end
+    end
+
+    def hesa_itt_data
+      equality_and_diversity_data = application_form&.equality_and_diversity
```

Comment:
we seem to be using `equality_and_diversity` here and for the fields above, but here we aren't protecting against it being `nil`, should we extract it to a method as so and reuse everywhere it occurs instead?
```
def equality_and_diversity
  application_form.equality_and_diversity || {}
end
```